Check that cached files still exist

With Gradle 6.7-rc-1, I've been getting this error occasionally:

```
Caused by: org.gradle.api.resources.MissingResourceException: Could not read script '/var/folders/7w/jcsr3q7106gbchv8gkvz09w40000gp/T/blowdryer-cache/https-raw.githubusercontent.com-mytakedo--dh4l6Q--src-main-resources-base-changelog.gradle' as it does not exist.
        at org.gradle.internal.resource.UriTextResource.getText(UriTextResource.java:147)
        at org.gradle.internal.resource.CachingTextResource.maybeFetch(CachingTextResource.java:112)
        at org.gradle.internal.resource.CachingTextResource.getContentHash(CachingTextResource.java:100)
        at org.gradle.groovy.scripts.internal.CrossBuildInMemoryCachingScriptClassCache.getOrCompile(CrossBuildInMemoryCachingScriptClassCache.java:43)
        at org.gradle.groovy.scripts.internal.BuildScopeInMemoryCachingScriptClassCompiler.compile(BuildScopeInMemoryCachingScriptClassCompiler.java:50)
        at org.gradle.groovy.scripts.DefaultScriptCompilerFactory$ScriptCompilerImpl.compile(DefaultScriptCompilerFactory.java:49)
        at org.gradle.configuration.DefaultScriptPluginFactory$ScriptPluginImpl.apply(DefaultScriptPluginFactory.java:110)
        at org.gradle.configuration.BuildOperationScriptPlugin$1.run(BuildOperationScriptPlugin.java:65)
```

The `/var/folders/7w/jcsr3q7106gbchv8gkvz09w40000gp/T/blowdryer-cache` folder exists, but it is empty.  The workaround is `./gradlew --stop`, and then it's fixed after that.

Seems like what's happening is that a long-lived daemon is caching "the script file exists", but macOS is clearing out the temp folder (as it is allowed to do).  So we need the local cache to check that the file exists before sending it.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -5,6 +5,8 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Fixed
+- Occasionally a file would be deleted from temp storage while a long-lived gradle daemon kept it in cache ([#11](https://github.com/diffplug/blowdryer/pull/18)).
 
 ## [1.1.0] - 2021-02-12
 ### Added

--- a/gradle.properties
+++ b/gradle.properties
@@ -3,7 +3,6 @@ maven_desc=Keep your gradle builds dry \u5E72
 maven_group=com.diffplug
 license=apache
 git_url=github.com/diffplug/blowdryer
-enable_publishing=true
 
 javadoc_links=\
     https://docs.oracle.com/javase/8/docs/api/ \

--- a/src/main/java/com/diffplug/blowdryer/Blowdryer.java
+++ b/src/main/java/com/diffplug/blowdryer/Blowdryer.java
@@ -80,7 +80,7 @@ public class Blowdryer {
 	public static File immutableUrl(String url) {
 		synchronized (Blowdryer.class) {
 			File result = urlToContent.get(url);
-			if (result != null) {
+			if (result != null && result.isFile()) {
 				return result;
 			}
 

--- a/src/test/java/com/diffplug/blowdryer/BlowdryerTest.java
+++ b/src/test/java/com/diffplug/blowdryer/BlowdryerTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2019-2020 DiffPlug
+ * Copyright (C) 2019-2021 DiffPlug
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -29,5 +29,13 @@ public class BlowdryerTest {
 
 	private void filenameSafe(String url, String safe) {
 		Assertions.assertThat(Blowdryer.filenameSafe(url)).isEqualTo(safe);
+	}
+
+	@Test
+	public void cachedFileDeleted_issue_11() {
+		String test = "https://raw.githubusercontent.com/diffplug/blowdryer/test/2/b/src/main/resources/sample";
+		Assertions.assertThat(Blowdryer.immutableUrl(test)).hasContent("b");
+		Blowdryer.immutableUrl(test).delete();
+		Assertions.assertThat(Blowdryer.immutableUrl(test)).hasContent("b");
 	}
 }

